### ISSUE TITLE
Mac M1: hyperkit not supported.

### ISSUE THREAD

**Issue by NitichaiSawangsai** [kubernetes/minikube#11885]
Mac M1

**Steps to reproduce the issue:** 

1. arch -x86_64 brew install hyperkit  
2. minikube start --vm=true
3. minikube start --driver=hyperkit

**Full output of failed command:**
<pre>
$ minikube start --driver=hyperkit <br/>
😄  minikube v1.21.0 on Darwin 11.4 (arm64)
✨  Using the hyperkit driver based on user configuration

❌  Exiting due to DRV_UNSUPPORTED_OS: The driver 'hyperkit' is not supported on darwin/arm64

</pre>

**Comment by spowelljr** [comment]
Hi @NitichaiSawangsai, thanks for reporting your issue!

As shown in the message, minikube does not yet support hyperkit on M1 Macs.

This is something we're hoping to support in the future. For now you can use the Docker driver.

Hope that helps, thanks for using minikube!

**Comment by sharifelgamal** [comment]
We're going to work on implementing a qemu driver for a vm driver on arm64 darwin.

**Comment by cedricjacobs** [comment]
> We're going to work on implementing a qemu driver for a vm driver on arm64 darwin.

Any clue when it is planned to be released?

**Comment by spowelljr** [comment]
> Any clue when it is planned to be released?

The current tentative timeline puts it the end of Q2 2022.

**Comment by ProbStub** [comment]
@spowelljr @sharifelgamal I totally love my M1 but my love for Docker is much more limited and I see the issue remains unassigned. Is there a recommended best practice as a workaround while this is being worked on, e.g. using podman?

**Comment by afbjorklund** [comment]
You can use lima, as a workaround.

**Comment by spowelljr** [comment]
We just released minikube v1.26.0-beta.0 that supports the QEMU driver `--driver=qemu2`.

https://github.com/kubernetes/minikube/releases/tag/v1.26.0-beta.0

**Comment by greenchapter** [comment]
> We just released minikube v1.26.0-beta.0 that supports the QEMU driver `--driver=qemu2`.
> 
> https://github.com/kubernetes/minikube/releases/tag/v1.26.0-beta.0

Looks on the first sight very good 👏

**Comment by sharifelgamal** [comment]
> @spowelljr is it possible to specify the path to qemu? I get
> 
> 🤷 Exiting due to PROVIDER_QEMU2_NOT_FOUND: The 'qemu2' provider was not found: lookup qemu: open /opt/homebrew/Cellar/qemu: no such file or directory
> 
> with an installation to /usr/local from source.
> 
> ```
> which qemu-system-aarch64
> /usr/local/bin/qemu-system-aarch64
> ```
> 
> (This is also qemu 7, so perhaps I should install qemu 6)

Not yet, this is something we're going to fix before the GA v1.26 release. Right now, we only look for a qemu installation from brew.

https://github.com/kubernetes/minikube/issues/14146 has more details about what's missing in the qemu driver.

**Comment by a1ex-var1amov** [comment]
Hi everyone, 
Has anybody else faced this with qemu2?

```
❯ minikube start --driver qemu2 --memory=4Gb
😄  minikube v1.26.0-beta.1 on Darwin 12.4 (arm64)
✨  Using the qemu2 (experimental) driver based on existing profile
❗  You cannot change the memory size for an existing minikube cluster. Please first delete the cluster.
👍  Starting control plane node minikube in cluster minikube
🔄  Restarting existing qemu2 VM for "minikube" ...
OUTPUT:
ERROR: qemu-system-aarch64: Addressing limited to 32 bits, but memory exceeds it by 1073741824 bytes
```

**Comment by sharifelgamal** [comment]
yeah, this is a known issue (seen in https://github.com/kubernetes/minikube/issues/14273) that should be fixed at HEAD, and will be fixed when the GA v1.26.0 minikube release is out.

**Comment by a1ex-var1amov** [comment]
@sharifelgamal thank you so much!

**Comment by leythadan** [comment]
> cluster

I got it to work by doing: 

`brew install qemu`

After the download, I did this:

`minikube start --driver qemu --memory=4Gb`

**Comment by greenchapter** [comment]
> I got it to work by doing:
> 
> `brew install qemu`
> 
> After the download, I did this:
> 
> `minikube start --driver qemu --memory=4Gb`

Thanks that works, but it uses qemu and not Hyperkit 😉

**Comment by afbjorklund** [comment]
HyperKit is Intel-only, so the alternative to QEMU will be the new Virtualization.framework driver (whenever that is ready)

* https://github.com/kubernetes/minikube/issues/12826

**Comment by greenchapter** [comment]
> HyperKit is Intel-only, so the alternative to QEMU will be the new Virtualization.framework driver (whenever that is ready)
> 
> * [Minikube with driver "new Virtualization Framework" on Apple M1 #12826](https://github.com/kubernetes/minikube/issues/12826)

Ya, you're totally right 👍

**Comment by rajjaiswalsaumya** [comment]
Can we please document this to the quickstart for mac ?

**Comment by dmiyamasu** [comment]
> Can we please document this to the quickstart for mac m1 ?

Has the quickstart documentation been started for mac m1? I don't see the updates published.

**Comment by greenchapter** [comment]
> > Can we please document this to the quickstart for mac m1 ?
> 
> Has the quickstart documentation been started for mac m1? I don't see the updates published.

@dmiyamasu and @rajjaiswalsaumya there is a Apple Silicon documentation here → https://minikube.sigs.k8s.io/docs/start/ and also the QEMU startup flags are explained on the drivers page → https://minikube.sigs.k8s.io/docs/drivers/qemu/

I think that will help.

**Comment by jgstew** [comment]
This document also only mentions hyperkit and doesn't mention using qemu on apple silicon: https://github.com/kubernetes/minikube/blob/master/site/content/en/docs/tutorials/docker_desktop_replacement.md

**Comment by nirs** [comment]
Hyperkit will be removed soon: #21601

/close
